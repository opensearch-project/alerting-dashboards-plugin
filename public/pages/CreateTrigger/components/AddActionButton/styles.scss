/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.add-action-button {
  &__flyout-button {
    width: 100%;
    height: 100%;
    min-height: 50px;
  }
}
