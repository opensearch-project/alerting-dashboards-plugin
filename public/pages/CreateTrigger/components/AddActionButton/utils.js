/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import _ from 'lodash';
import { DEFAULT_MESSAGE_SOURCE, FORMIK_INITIAL_ACTION_VALUES } from '../../utils/constants';
import { MONITOR_TYPE } from '../../../../utils/constants';
import { getDigitId, getUniqueName } from '../../../../utils/helpers';

export const getInitialActionValues = ({ monitorType, flyoutMode, actions }) => {
  const initialActionValues = _.cloneDeep(FORMIK_INITIAL_ACTION_VALUES);

  switch (monitorType) {
    case MONITOR_TYPE.BUCKET_LEVEL:
      _.set(
        initialActionValues,
        'message_template.source',
        DEFAULT_MESSAGE_SOURCE.BUCKET_LEVEL_MONITOR
      );
      break;
    case MONITOR_TYPE.QUERY_LEVEL:
    case MONITOR_TYPE.CLUSTER_METRICS:
      _.set(
        initialActionValues,
        'message_template.source',
        DEFAULT_MESSAGE_SOURCE.QUERY_LEVEL_MONITOR
      );
      break;
  }

  const id = getDigitId();
  initialActionValues.id = `notification${id}`;

  // Add name based on previous name;
  if (flyoutMode) {
    initialActionValues.name = getUniqueName(actions, 'Notification ');
  }

  return initialActionValues;
};
