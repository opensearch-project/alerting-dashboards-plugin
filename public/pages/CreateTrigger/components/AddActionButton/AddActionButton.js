/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import _ from 'lodash';
import { EuiButton, EuiButtonEmpty, EuiPanel } from '@elastic/eui';
import { getInitialActionValues } from './utils';
import { MONITOR_TYPE } from '../../../../utils/constants';
import './styles.scss';

const AddActionButton = ({
  arrayHelpers,
  type = 'slack',
  numActions,
  flyoutMode,
  onPostAdd,
  values,
  fieldPath,
}) => {
  const buttonText =
    numActions === undefined || numActions === 0 ? 'Add action' : 'Add another action';
  const monitorType = _.get(arrayHelpers, 'form.values.monitor_type', MONITOR_TYPE.QUERY_LEVEL);
  const onClick = () => {
    const actions = _.get(values, `${fieldPath}actions`, []);
    const initialValues = getInitialActionValues({ monitorType, flyoutMode, actions });
    arrayHelpers.push(initialValues);

    if (onPostAdd) {
      onPostAdd(initialValues);
    }
  };

  return flyoutMode ? (
    <EuiPanel paddingSize="none">
      <EuiButtonEmpty
        onClick={onClick}
        iconType="plusInCircle"
        className="add-action-button__flyout-button"
      >
        Add notification
      </EuiButtonEmpty>
    </EuiPanel>
  ) : (
    <EuiButton fill={false} size={'s'} onClick={onClick}>
      {buttonText}
    </EuiButton>
  );
};

export default AddActionButton;
