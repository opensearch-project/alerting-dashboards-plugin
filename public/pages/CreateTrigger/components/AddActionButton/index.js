/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import AddActionButton from './AddActionButton';

export default AddActionButton;
