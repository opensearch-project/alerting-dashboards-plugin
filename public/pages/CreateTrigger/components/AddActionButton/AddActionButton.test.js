/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow } from 'enzyme';
import AddActionButton from './AddActionButton';

describe('AddActionButton', () => {
  test('renders', () => {
    const wrapper = shallow(<AddActionButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
