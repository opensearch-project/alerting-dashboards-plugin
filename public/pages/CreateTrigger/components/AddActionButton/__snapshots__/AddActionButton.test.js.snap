// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddActionButton renders 1`] = `
<EuiButton
  fill={false}
  onClick={[Function]}
  size="s"
>
  Add action
</EuiButton>
`;
